docs/ldml/tr35-numbers.md

```diff
@@ -1315,6 +1321,73 @@ Here is a set of heuristic rules that may be helpful:
+
+*Approximate number formatting* refers to a specific format of numbers in which the value is understood to not be exact; for example, "\~5 minutes".
+
+To format an approximate number, follow the normal number formatting procedure in Number Format Patterns](#Number_Format_Patterns), but substitute the `approximatelySign` from [Number Symbols](#Number_Symbols) in for the minus sign placeholder.
+
+If the number is negative, or if the formatting options request the sign to be displayed, *prepend* the `approximatelySign` to the plus or minus sign before substituting it into the pattern. For example, "\~-5" means "approximately negative five". This procedure may change in the future.
+
+### 8.2 <a name="Collapsing_Number_Ranges" href="#Collapsing_Number_Ranges">Collapsing Number Ranges</a>
+
+*Collapsing* a number range refers to the process of removing duplicated information in the *lower* and *upper* values. For example, if the lower string is "3.2 centimeters" and the upper string is "4.5 centimeters", it is desirable to remove the extra "centimeters" token.
+
+This operation requires *semantic annotations* on the formatted value. The exact form of the semantic annotations is implementation-dependent. However, implementations may consider the following broad categories of tokens:
+
+1. Numerical value, including decimal and grouping separators
+1. Sign symbol
+1. Scientific or compact notation
+1. Unit of measurement
+
+For example, consider the string `-5.3M US dollars`. It may be annotated as follows:
+
+- `-` → sign symbol
+- `5.3` → numerical value
+- `M` → compact notation
+- `US dollars` → unit of measurement for the currency USD
+
+Two tokens are *semantically equivalent* if they have the same *semantic annotations*, even if they are not the exact same string. For example:
+
+1. "centimeter" is semantically equivalent to "centimeters".
+1. "K" (the thousands symbol in compact decimals) is NOT semantically equivalent to "K" (the measurement unit Kelvin).
+
+Two semantically equivalent tokens can be *collapsed* if they appear at the start of both values or the end of both values. However, the implementation may choose different levels of aggressiveness with regard to collapsing tokens.
+
+**Heuristic:** Only collapse semantically equivalent *unit of measurement* tokens, and only if the tokens are more than one code point in length. This is to avoid ambiguous strings such as "3–5K" (could represent 3–5000 or 3000–5000).
+
+**To collapse tokens:** Remove the token from both values, and then re-compute the token based on the number range. If the token depends on the plural form, follow [Plural Ranges](#Plural_Ranges) to calculate the correct form. If the tokens originated at the beginning of the string, prepend the new token to the beginning of the *lower* string; otherwise, append the new token to the end of the *upper* string.
+
```

Comment:
Add:
The above description describes the expected output. Internally, the implementation may determine the equivalent units of measurement by passing the codes back from the number formatters, allowing for a precise determination of "semantically equivalent".

Reply:
Added, but a few lines above where you suggested